doc: Enhance documentation section on labeling

amarkovicsa opened the pull request:
Mostly due to the limitations of table support in rst file format, the current content of the section "Labeling issues and pull requests in Github" is difficult to navigate and follow, especially for new contributors, and especially in pdf output..

Fix this by rearranging the content of the section and fixing some minor discrepancies between Github and documentation in the section.

Note that this pull request does only minor changes regarding the information contained in the section. It mostly rearranges it and makes it more accessible to the reader.

amarkovicsa committed:
doc: Rearrange content of section "Labeling issues and pull requests"

Mostly due to limited support for tables in rst format, information on
labels used for Zephyr issues and pull requests in Github is hard to
navigate and is visually not pleasing. Rearange the content of the
section so that bullets are mainly utilized rather than tables. Format
label names as 'emphasis', rather that 'monospace' (which should be
used primarily for code snippets, and Github labels are not code).
Also, group the information according to applicability (taking into
account if a label is applicable to:

  * issues only;
  * pull requests only;
  * both issues and pull requests.

Label names and decriptions are otherwise left 'as is' in this commit.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

carlescufi commented:
I will take a look at this after Easter.

amarkovicsa commented:
> I will take a look at this after Easter.

Thanks a lot!

mbolivar-nordic reviewed (commented):
I think I understand the problem and agree it is worth solving it, but I do not think that the current state of the PR should be merged.

Keeping the lists of labels separate from their descriptions will make it easier for them to go stale. For example, if I add a label applicable to both PRs and issues, but forget to add a description for some reason, the docs will be stale. This is harder to do in the current documentation, where the descriptions are right next to the labels themselves.

Why not split this into 3 sections:

1. Labels applicable to issues: list of labels with descriptions
2. Labels applicable to PRs only: list of labels with their descriptions
3. Labels applicable to both: list of labels, perhaps with descriptions from previous sections 

That should avoid issues with PDF rendering but still keep this relatively easy to maintain.

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -258,138 +258,165 @@ fix it.
+Labels applicable to both pull requests and issues
+==================================================
+
+* ``Area:*``
```
This is definitely wrong in main, but the fix is also not correct. It should be `area: *` (lowercase a, with space).

amarkovicsa replied:
Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/cab8ea2758bb3f11808fc10b046d7cfa5d797508 and https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/52e4fca93ec5025eebfd4c151ea3b878b4e82766 .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -258,138 +258,165 @@ fix it.
+An area maintainer should be able to filter by an area label and find all issues
+and PRs which relate to that area.
+
+* ``Platform:*``
```
This should be `platform: *`.

amarkovicsa replied:
Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/cab8ea2758bb3f11808fc10b046d7cfa5d797508 .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -263,7 +263,7 @@ These are the labels we currently have, grouped by applicability:
 ================================
 
-* ``priority: {high|medium|low}``
+* ``priority:{high|medium|low}``
```
I think this is not correct. I see a space on GitHub. Here is an example search: https://github.com/zephyrproject-rtos/zephyr/issues?q=is%3Aissue+is%3Aopen+label%3A%22priority%3A+high%22

amarkovicsa replied:
Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/cab8ea2758bb3f11808fc10b046d7cfa5d797508 .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -347,27 +350,34 @@ The PR is a backport or should be backported.
-Indicates subsystems (e.g., Kernel, I2C, Memory Management), project functions
-(e.g., Debugging, Documentation, Process), or other categories
-(e.g., Coding Style, MISRA-C) affected by the bug or pull request.
+Indicates subsystems (e.g., "Area: Kernel", "Area: I2C", "Area: Memory
+Management"), project functions (e.g., "Area: Debugging", "Area: Documentation",
+"Area: Process"), or other categories (e.g., "Area: Coding Style", "Area:
+MISRA-C") affected by the bug or pull request.
```
Lowercase "a" throughout this sentence.

amarkovicsa replied:
This is now fixed in a separate commit https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/52e4fca93ec5025eebfd4c151ea3b878b4e82766 .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -347,27 +350,34 @@ The PR is a backport or should be backported.
 
-The issue is to be discussed in the following `dev-review/TSC meeting`_ if time
-permits.
+The issue or PR is to be discussed in the following `dev-review/TSC meeting`_ if
+time permits.
```
Did you mean to re-word this to just say TSC?

amarkovicsa replied:
Yes, you guessed right, that was my intention. It is now fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/0fd42484f58a06739a1b4c731bf80c76e8138d8d .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -378,7 +388,7 @@ in :ref:`stable_api_changes`.
 
 * ``Bug``
 
-The issue is a bug, or the PR is fixing a bug       |
```
This typo was introduced in this pull request and therefore the fix should be squashed into the previous commit instead.

amarkovicsa replied:
Fixed in new version of this PR.

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -315,23 +315,26 @@ Labels applicable to pull requests only
 * ``Security Review``
 
-Depending on the PR complexity, an indication of how long a merge should be held
-to ensure proper review. See :ref:`review process <review_time>`.
```
This was also a change introduced in this pull request, so it's not clear why this is being done in this commit. I'm also not sure the `Licensing` label description is a good place for this to end up.

amarkovicsa replied:
I slightly reworded the sentence and put it as note at the end of "Labels applicable to PRs only" subsection. I think it is now more clear and reflect IMO better the original intent. What do you think, @mbolivar-nordic ?

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -258,138 +258,155 @@ fix it.
+
+Long term release branch related.
+
+* ``EXT``
+
+Related to an external component (in ``ext/``).
```
Since you are fixing discrepancies, do you mind removing this one as well? We no longer use ext/; this directly was removed after we moved most external components to separate modules.

amarkovicsa replied:
Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/ef616ef18c05d6fb36dcd63a13838f418df98e21 .

mbolivar-nordic commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -258,138 +258,165 @@ fix it.
+Labels applicable to pull requests only
+=======================================
+
+The issue or PR describes a change to a stable API. See additional information
+in :ref:`stable_api_changes`.
```
This doesn't belong here. This is a description of the ``Stable API Change`` label. It should be removed from here.

amarkovicsa replied:
I think it is now fixed in https://github.com/zephyrproject-rtos/zephyr/pull/44774/commits/a35f8a0c5d4be4044ed1036eb1603db4c6c2c352 .

mbolivar-nordic reviewed (commented):
Thanks very much for the update. The new structure seems reasonable to me personally. I looked at the details and found a few issues.

amarkovicsa commented:
> Thanks very much for the update. The new structure seems reasonable to me personally. I looked at the details and found a few issues.

I truly appreciate all your comments and hints. I will continue working on this PR after the holiday. @mbolivar-nordic

amarkovicsa committed:
doc: Fix label names to be in sync with Github label names

Some label names in the documentation do not match with what is
seen in Github interface. Fix it by syncing with Github.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Move subsection on "Stale" label to the right place

Label "Stale" is used for both issues and pull requests. It is currenly
listed as applicaple to pull requests only, however. Move it to the
right place.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Remove reference to now non-existant ext subdirectory

Directory <Zephyr-root>/ext does not exist anymore. Remove the
reference to it from the documentation.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Split and enhance subsections on "dev-review" and "TSC" labels

Split the explanation on dev-review and TSC labels into two separate ones,
and slightly reword both. For both, amend corresponding link to point to
the more relevant place in the online page on Zephyr meetings. Add
explanation for acronym "TSC".

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Enhance description on "area: *" label

Use full label names for examples in the description, and other minor
enhancements.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Enhance subsection on "Stable API Change" label

Remove part that it is not appropriate.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Enhance subsection on "Hotfix" label

Sync the text of description with what is seen in Github.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Enhance subsection on "Maintainer" label

Sync the text of description with what is seen in Github.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Enhance subsection on "Security Review" label

Sync the text of description with what is seen in Github.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

amarkovicsa committed:
doc: Reword and reformat a sentence on review time

Move the sentence to the end of the section on labels applicable to
PRs only, reword it and convert to a note, indicating that the note
is for all labels in this section.

Signed-off-by: Aleksandar Markovic <aleksandar.markovic.sa@gmail.com>

mbolivar-nordic reviewed (approved):
@carlescufi this looks OK to me; could you please take a look?

Thanks @amarkovicsa

carlescufi commented on doc/project/dev_env_and_tools.rst:
```diff
@@ -258,138 +258,170 @@ fix it.
+
+Something, which was working, but does not anymore (bug subtype).
+
+* *Question*
```
We removed this label, but OK to fix this in a follow-up
